frappe/model/naming.py

```diff
@@ -165,16 +165,7 @@ def set_new_name(doc):
 	if not doc.name and autoname:
 		set_name_from_naming_options(autoname, doc)
 
-	# if the autoname option is 'field:' and no name was derived, we need to
-	# notify
-	if not doc.name and autoname.startswith("field:"):
-		fieldname = autoname[6:]
-		frappe.throw(_("{0} is required").format(doc.meta.get_label(fieldname)))
-
```

Comment:
moved this to `set_name_from_naming_options` so that we don't verify if the autoname is `field:` again